hazelcast-jet-core/src/main/java/com/hazelcast/jet/Sinks.java

```diff
@@ -58,6 +64,204 @@ private Sinks() {
+     *  @param toValueFn  function that extracts the value from the input item
+     *  @param mergeFn  function contains the merge logic. The function
+     *                 will be called with {@code mergeFn.apply(V oldValue, V newValue)}
+     *  @param <E> type of input item
+     *  @param <K> type of the key
+     *  @param <V> type of the value
+     */
+    public static <E, K, V> Sink<E> mergingMap(
+            @Nonnull String mapName,
+            @Nonnull DistributedFunction<E, K> toKeyFn,
+            @Nonnull DistributedFunction<E, V> toValueFn,
+            @Nonnull DistributedBinaryOperator<V> mergeFn
+    ) {
+        return fromProcessor("mergingMapSink(" + mapName + ')',
+                mergeMapP(mapName, toKeyFn, toValueFn,  mergeFn));
+    }
+
+    /**
+     * Returns a sink that updates the items it receives into a Hazelcast
+     * {@code IMap} with the specified name and merge function in a
+     * remote cluster identified by the supplied {@code ClientConfig}.
+     * The value extracted from the input item will be inserted into the map
+     * if there is no value associated with the key, otherwise {@code mergeFn}
+     * will be called with the associated value as {@code oldValue}  and the value which
+     * we want insert as {@code newValue}. The return value of the {@code mergeFn}
+     * will be the new value of the key.
+     *
+     *  @param mapName name of the map
+     *  @param clientConfig client configuration to be able to connect remote
+     *                      cluster
+     *  @param toKeyFn  function that extracts key from the input item
+     *  @param toValueFn  function that extracts value from the input item
+     *  @param mergeFn  function contains the merge logic. The function
+     *                 will be called with {@code mergeFn.apply(V oldValue, V newValue)}
+     *  @param <E> type of input item
+     *  @param <K> type of the key
+     *  @param <V> type of the value
+     */
+    public static <E, K, V> Sink<E> remoteMergingMap(
+            @Nonnull String mapName,
+            @Nonnull ClientConfig clientConfig,
+            @Nonnull DistributedFunction<E, K> toKeyFn,
+            @Nonnull DistributedFunction<E, V> toValueFn,
+            @Nonnull DistributedBinaryOperator<V> mergeFn
+    ) {
+        return fromProcessor("remoteMergingMapSink(" + mapName + ')',
+                mergeRemoteMapP(mapName, clientConfig, toKeyFn, toValueFn, mergeFn));
+    }
+
+    /**
+     * Returns a sink that updates the items it receives into a Hazelcast
+     * {@code IMap} with the specified name and update function. The function
+     * will be wrapped inside an {@link EntryProcessor}. The {@code updateFn}
+     * will be called with the input item as item and the {@code oldValue}, which is
+     * the associated value of the extracted key in Hazelcast Map. The return
+     * value of the {@code updateFn} will be the new value of the key.
```

Comment:
see my comments about mergingMap. It's not 100% clear how the update logic works.